federation/cmd/kubefed/app/kubefed.go

```diff
@@ -34,7 +35,7 @@ const (
 )
 
 func GetDefaultServerImage() string {
-	return fmt.Sprintf("%s:%s", hyperkubeImageName, version.Get())
+	return fmt.Sprintf("%s:%s", hyperkubeImageName, strings.Replace(version.Get().String(), "+", "_", 1))
```

Comment:
Are there other chars that will show up that will need special handling? Is using a standard mechanism like the url encoding be useful here?

Reply:
@dims, AFAIK there will be no other characters which need special handling. Also note that the docker image tag should follow the naming convention as below: (taken from the docker documentation)
```
A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes. A tag name may not start with a period or a dash and may contain a maximum of 128 characters.
```
Since our `version.Get()` returns an image with '+' character, we chose to replace it with '_' character in [here](https://github.com/kubernetes/kubernetes/blob/master/build/util.sh#L31), while pushing the docker image to gcr in [here](https://github.com/kubernetes/kubernetes/blob/master/federation/develop/develop.sh#L118:L136)

Reply:
LGTM 👍